libs/execution_context/ebpf_link.c

```diff
@@ -154,10 +164,23 @@ _ebpf_link_extension_changed_callback(
 
     // Complete detach.
     if (provider_data == NULL) {
+        // Check if the link is in the process of attaching.
+        if (link->state == EBPF_LINK_STATE_DETACHING) {
+            // If so, complete the detach.
+            _ebpf_link_set_state(link, EBPF_LINK_STATE_DETACHED);
+        } else if (link->state == EBPF_LINK_STATE_ATTACHED) {
+            // If not, mark the link as ready to reattach.
+            _ebpf_link_set_state(link, EBPF_LINK_STATE_ATTACHING);
```

Comment:
This seems to be in the 'detach' path, so shouldn't this be `EBPF_LINL_STATE_DETACHING`? Same for the comment in the line above.

Reply:
resolving per 1/1 discusion.